src/e2e/puppeteer/__tests__/editable-gap.ts

```diff
@@ -0,0 +1,116 @@
+import clickThought from '../helpers/clickThought'
+import getEditable from '../helpers/getEditable'
+import getEditingText from '../helpers/getEditingText'
+import getSelection from '../helpers/getSelection'
+import hideHUD from '../helpers/hideHUD'
+import paste from '../helpers/paste'
+import { page } from '../setup'
+
+// Intentional overlap between thoughts to fix selection behavior
+// See: useSizeTracking.ts lineHeightOverlap calculation
+// const EXPECTED_OVERLAP = DEFAULT_FONT_SIZE / 8
+
+vi.setConfig({ testTimeout: 20000, hookTimeout: 20000 })
+
+/** Helper function to check if cursor is positioned in the middle of a thought. */
+async function isCursorInMiddle(expectedThought: string): Promise<boolean> {
+  const editingText = await getEditingText()
+  if (editingText !== expectedThought) return false
```

Comment:
I probably wouldn't bundle the editing text value assertion into `isCursorInMiddle`. Were you doing that for efficiency? I think it's probably better to let the function stick to its main purpose.